### DISCUSSION TITLE
STELLA 2.0: Using CircuitPython 9 and Updated Feather nRF52840 Sense

### DISCUSSION THREAD

**Discussion by Whitw-pers** [STELLA-Landsat/STELLA#57] [Build Notes: How do I make a STELLA?]
I recently started building a few STELLA instruments for future student projects at my university, and I encountered some issues while trying to program them that I want to document here for anyone else building a STELLA 2.0 since January of this year. 

When I initially copied the code file and lib folder from S2 Software onto the CIRCUITPY drive and ran in the Mu editor, I got the error message
`  File "code.py", line 29, in <module>
ValueError: incompatible .mpy file`

The solution turned out to be downloading the [CircuitPython 9.x library bundles](https://circuitpython.org/libraries) and replacing the libraries in the existing lib folder with their 9.x versions. Certain libraries that used to just be a .mpy file are now folders, including the adafruit_pcf8523 library. I changed line 32 in code.py to `from adafruit_pcf8523 import pcf8523` to adjust for that.

Now, when I ran code.py I got the output
`  File "code.py", line 1118, in <module>
  File "code.py", line 107, in main
  File "code.py", line 729, in initialize_MOT_sensor
  File "adafruit_lsm6ds/__init__.py", line 230, in __init__
RuntimeError: Failed to find LSM6DS33 - check your wiring!`

It turns out that as of 1/10/24 Adafruit replaced the LSM6DS33 sensor on the Feather Sense with the LSM6DS3TR-C sensor. My solution was to CTRL+F all instances of lsm6ds33 and LSM6DS33 and replace them with lsm6ds3trc and LSM6DS3TRC, respectively. This now seems to work, my STELLA 2.0 devices seem to run just fine now.

Here is the new code.py file and lib folder I used for anyone else working with CircuitPython 9 and the new Feather Sense board:
[CPY_S2_Code_and_Libraries.zip](https://github.com/STELLA-Landsat/STELLA/files/14890659/CPY_S2_Code_and_Libraries.zip)

For anyone else who found themselves stuck while making a STELLA 2.0, good luck and I hope this is helpful!

**Comment by paulmirel** [comment]
Thank you so much for finding and fixing that problem, and for documenting your solution so well!

**Comment by Quent1Dhn** [comment]
Using that previous fix, I could run the code itself. However, if we want to use the test code for the clock : _04. real time clock interactive console_ then a slight change needs to be made.
Here's line 7 to 18 of what I ran successfully to set up the clock : 
```
import busio
from adafruit_pcf8523 import pcf8523 # this line changed using @Whitw-pers suggestion
import time
#import rtc    # import the microcontroller system real time clock functionality
import board
import displayio

displayio.release_displays()

i2c_bus = busio.I2C(board.SCL, board.SDA)
pcf_rtc = pcf8523.PCF8523(i2c_bus)    # this line has to be modified too because the import has changed and PCF8523 wouldn't not be recognized
#rtc.set_time_source(pcf_rtc)    # set the time source for the microcontroller real time clock functionality to the real time clock hardware object

```
This is what I used to have the correct time clock with the right imports. 
Hope this helps !
